Components/Order/Factory/DetailFactory.php

```diff
@@ -50,8 +50,8 @@ public function create(PositionStruct $positionStruct, $isTaxFree)
             throw new InvalidOrderException('No product number was passed.');
         }
 
-        if ($positionStruct->isDiscount() || $positionStruct->isSurcharge()) {
-            return $this->createDiscount($positionStruct);
+        if ($positionStruct->isDiscount()) {
+            return $this->createDiscount($positionStruct, $isTaxFree);
```

Comment:
After your rebase you no longer check for $positionStruct->isSurcharge(). I don't think this was intentional ;) Also please verify the correct functionality of this pullrequest. @nicoanders

Reply:
You´re completly right :) sorry... my fault. I resolve it and squash the commits